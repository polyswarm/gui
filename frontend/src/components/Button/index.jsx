// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty Manager imports
// Component imports
import './strings.js';

class Button extends Component {
  constructor(props) {
    super(props);
  }

  render() {

  }

}
Button.proptypes = {

}
export default Button;