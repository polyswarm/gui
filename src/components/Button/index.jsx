// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

class Button extends Component {
  constructor(props) {
    super(props);
    this.onClickHandler = this.onClickHandler.bind(this);
  }

  render () {
    const { props: { disabled, children, cancel, flat } } = this;
    const computedClass = classNames('Button', {
      'cancel': !flat && cancel,
      'flat': flat && !cancel,
      'flat-cancel': flat && cancel
    });
    return (
      <button
        disabled={disabled}
        className={computedClass}
        onClick={this.onClickHandler}>
        {children}
      </button>
    );
  }

  onClickHandler() {
    const { props: { onClick } } = this;
    if (onClick) {
      onClick();
    }
  }
}

Button.proptypes = {
  onClick: PropTypes.func.isRequired,
  cancel: PropTypes.bool,
};

export default Button;
