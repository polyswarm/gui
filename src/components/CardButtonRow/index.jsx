// Vendor Imports
import React, {Component} from 'react';
import PropTypes from 'prop-types';
// Bounty Management Imports

// Component Imports

class CardButtonRow extends Component {
    render() {
        const {props: {children}} = this;
        return (
            <div className='CardButtonRow'>
                {React.Children.map(children, (child) => {
                  return (
                    <div
                      key={child.textconent}
                      className='ButtonWrap'>
                      {child}
                    </div>
                  ); 
                })}
            </div>
        );
    }
}
CardButtonRow.proptypes = {
    title: PropTypes.string,
}
export default CardButtonRow;