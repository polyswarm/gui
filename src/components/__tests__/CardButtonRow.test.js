import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import CardButtonRow from '../CardButtonRow';
import Button from '../Button';

it('renders without crashing', () => {
  const wrapper = render(<CardButtonRow />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('shows the children passed', () => {
  const wrapper = render(
    <CardButtonRow>
      <Button>Child</Button>
    </CardButtonRow>
  );

  expect(wrapper.find('button').text()).toBe('Child');

  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('shows multiple children when passed', () => {
  const wrapper = mount(
    <CardButtonRow>
      <Button>Child</Button>
      <Button>Second</Button>
    </CardButtonRow>
  );

  expect(wrapper.find('button')).toHaveLength(2);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

