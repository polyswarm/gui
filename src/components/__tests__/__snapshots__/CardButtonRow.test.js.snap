// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="CardButtonRow"
>
  
</div>
`;

exports[`shows multiple children when passed 1`] = `null`;

exports[`shows the children passed 1`] = `
<div
  class="CardButtonRow"
>
  <div
    class="ButtonWrap"
  >
    <button
      class="Button"
    >
      Child
    </button>
  </div>
</div>
`;
